common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -1143,6 +1143,378 @@ def verify_get_parent_locations_results(self, expected_results):
                 self.store.get_parent_location(child_location, revision=revision)
             )
 
+    def verify_item_parent(self, item_location, expected_parent_location, old_parent_location, is_reverted=False):
+        """
+        Verifies that item is placed under expected parent.
+
+        Arguments:
+            item_location (BlockUsageLocator)    : Locator of item.
+            expected_parent_location (BlockUsageLocator)  : Expected parent block locator.
+            old_parent_location (BlockUsageLocator)  : Old parent block locator.
+            is_reverted (Boolean)   : A flag to notify that item was reverted.
+        """
+        source_item = self.store.get_item(item_location)
+        old_parent = self.store.get_item(old_parent_location)
+        expected_parent = self.store.get_item(expected_parent_location)
+
+        self.assertEqual(expected_parent_location, source_item.get_parent().location)
+
+        # If an item is reverted, it means it's actual parent was the one that is the current parent now
+        # i.e expected_parent_location otherwise old_parent_location.
+        published_parent_location = expected_parent_location if is_reverted else old_parent_location
+
+        # Check parent locations wrt branches
+        with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+            self.assertEqual(expected_parent_location, self.store.get_item(item_location).get_parent().location)
+        with self.store.branch_setting(ModuleStoreEnum.Branch.published_only):
+            self.assertEqual(published_parent_location, self.store.get_item(item_location).get_parent().location)
+
+        # Make location specific to published branch for verify_get_parent_locations_results call.
+        published_parent_location = published_parent_location.for_branch(ModuleStoreEnum.BranchName.published)
+
+        # Verify expected item parent locations
+        self.verify_get_parent_locations_results([
+            (item_location, expected_parent_location, None),
+            (item_location, expected_parent_location, ModuleStoreEnum.RevisionOption.draft_preferred),
+            (item_location, published_parent_location, ModuleStoreEnum.RevisionOption.published_only),
+        ])
+
+        # Also verify item.parent has correct parent location set.
+        self.assertEqual(source_item.parent, expected_parent_location)
+        self.assertEqual(source_item.parent, self.store.get_parent_location(item_location))
+
+        # Item should be present in new parent's children list but not in old parent's children list.
+        self.assertIn(item_location, expected_parent.children)
+        self.assertNotIn(item_location, old_parent.children)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_update_item_parent(self, default_ms):
```

Comment:
`default_ms ` is used at most places in this file as well others. But I can use `store_type` which is now used mostly to know store type.